### ISSUE TITLE
[Bug]: MultiSelectBuilder when string of a filter is given errors out

### ISSUE THREAD

**Issue by racerguy** [Power-Components/livewire-powergrid#1059]
### Have you searched through other issues to see if your problem is already reported or has been fixed?

Yes, I did not find it.

### Did you read the documentation?

Yes, I did not find it.

### Have you tried to publish the views?

Yes - I didn't work.

### Is there an error in the console?

No

### PHP Version

8.2

### PowerGrid

4.8.5

### Laravel

10

### Theme

Tailwind 3.x

### Describe the bug.

When using a join with MultiSelect.  "count(): Argument #1 ($value) must be of type Countable|array, string given"

Occurs due to 
count($values ) Line 25  in WithMultiSelectBuilder

Values are allowed to be array|int|string|null. However, count only works on array

### To Reproduce...

First click on "FOO" then....

### Extra information

```markdown
<?php
 //...
```

**Comment by racerguy** [comment]
My relationship tree is like this:

Orders have customers (but customers also have salesperson)
Orders have salesperson

When using salesperson_id I get the error that salesperson_id is ambiguous because it is on the primary Orders model and on customer. 

The fix of this would be orders.salesperson_id in the data_field however, I get this error.

**Comment by racerguy** [comment]
Regular Select Output from DD($column) in builder.php

`array:1 [▼ // vendor/power-components/livewire-powergrid/src/Helpers/Builder.php:79
  "orders" => array:1 [▼
    "salesperson_id" => array:1 [▼
      0 => "3"
    ]
  ]
]`


DD using multiselect 

`array:1 [▼ // vendor/power-components/livewire-powergrid/src/Helpers/Builder.php:79
  "orders" => array:1 [▼
    "salesperson_id" => array:1 [▼
      0 => "3"
    ]
  ]
]
`

**Comment by racerguy** [comment]
Arr:dot() seems to be the problem in builder its flattening the value array (last array in dimension).

Below is a snippet of a temp fix I did to create a customDot implementation to fix it and seems to be working on my side. 


```````
public static function CustomDot($array, $prepend = '')
    {
        $results = [];

        foreach ($array as $key => $value) {
            if (is_array($value) && ! empty($value)  ) {
                if(is_numeric(array_key_first($value))){
                    $results[$prepend.$key] = $value;
                    break;
                }
                $results = array_merge($results, static::CustomDot($value, $prepend.$key.'.'));
            } else {
                $results[$prepend.$key] = $value;
            }
        }

        return $results;
    }

````````

**Comment by luanfreitasdev** [comment]
Fixed on v4.8.6
